controllers/kustomization_controller.go

```diff
@@ -80,6 +80,17 @@ func (r *KustomizationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 		return ctrl.Result{}, err
 	}
 
+	if source.GetArtifact() == nil {
+		msg := "Source is not ready"
+		kustomization = kustomizev1.KustomizationNotReady(kustomization, kustomizev1.ArtifactFailedReason, msg)
+		if err := r.Status().Update(ctx, &kustomization); err != nil {
+			log.Error(err, "unable to update Kustomization status")
+			return ctrl.Result{Requeue: true}, err
+		}
+		log.Info(msg)
+		return ctrl.Result{}, nil
```

Comment:
Any specific reason for using `Requeue: true` in the `if`-statement above this but not here?

Reply:
Yes, requeueing here is useless since the artifact is not ready, while a status update error could be caused be a transient network error

Reply:
"Not ready" gives me the idea that it is transient as well, and the resource is still initializing?

Reply:
It can take forever

Reply:
That’s why it uses a exponential backoff strategy?

Reply:
Maybe it’s not obvious from these changes but once the git repo is initialized and the artifact is ready, the kustomizion is triggered right away, there is no reason to spam the api server with calls while source-controller is cloning the repo.

Reply:
Ah, I forgot about that gear wheel. Ack.